modules/event-emitter/main/index.coffee

```diff
@@ -74,19 +74,23 @@ class EventEmitter
     ee.emitters.add(be)
     be
 
-  removeEmitter = (ee, namespace) ->
-    if ee.emittersMap.has(namespace)
+  removeEmitter = (ee, be, namespace) ->
+    if namespace and ee.emittersMap.has(namespace)
       ee.emittersMap.delete(namespace)
-      ee.emitters.remove(ee)
-    ee
+    else
+      lookedUpNamespace = ee.emittersMap.entries().filter(([_, e]) -> e is be)[0]?[0]
+      if lookedUpNamespace
+        ee.emittersMap.delete(lookedUpNamespace)
+    ee.emitters.remove(be)
```

Comment:
Updated `removeEmitter` to take the basic emitter you want to remove and updated to look for the emitter in the namespaced list as otherwise `.off()` doesn't behave properly, leaving emitters in the map but removing them from the `emitters`.
This function was previously unused...